src/PiGui.cpp

```diff
@@ -284,7 +284,7 @@ void *PiGui::makeTexture(const std::string &filename, unsigned char *pixels, int
 	Pi::renderer->CheckRenderErrors(__FUNCTION__, __LINE__);
 	// nasty bit as I invoke the TextureGL
 	Graphics::OGL::TextureGL *pGLTex = reinterpret_cast<Graphics::OGL::TextureGL*>(pTex);
-	Uint32 result = pGLTex->GetTexture();
+	Uint32 result = pGLTex->GetTextureID();
```

Comment:
I think we could switch from `Uint32` to `uint32_t`, and their ilk.
I've heard that `std::shared_ptr` has a [lot of performance](http://stackoverflow.com/questions/3628081/shared-ptr-horrible-speed)([2](http://blog.davidecoppola.com/2016/10/performance-of-raw-pointers-vs-smart-pointers-in-cpp/) and [usage issues](http://seanmiddleditch.com/dangers-of-stdshared_ptr/) so it might be worth benchmarking our `RefCountedPtr` vs `std::shared_ptr`.